firebase/src/core/domain/models/collection.ts

```diff
@@ -5,10 +5,10 @@ import * as Joi from "joi";
 interface Collection {
   readonly id: string;
   readonly name: string;
+  readonly description: string;
+  readonly tags: string[];
+  readonly category: string;
 
-  readonly tags?: string[];
-  readonly category?: string;
-  readonly description?: string;
```

Comment:
Updated these properties because, since we are talking about a public collection they should be required (at least the description should) IMO.

Reply:
But then why is it required in the `Collection` interface? Public do indeed require ALL properties, but personal collections should not require anything other than an `id`, `name` and at least a single memo. We don't have a class representation for these in the server as we have no need for these up until now.

Reply:
But then why is it required in the `Collection` interface? Public do indeed require ALL properties, but personal collections should not require anything other than an `id`, `name` and at least a single memo. We don't have a class representation for these in the server as we have no need for these up until now.

Reply:
I didn't get your point here, do you agree or disagree? `Public` collections should have such properties as required and `Private` not, but since we don't have private Collections in the server yet, this is not in this PR scope, right?

Reply:
We can take advantage from TS and override them to become required only in the `PublicCollection`. Updated the interfaces. Let me know what do you think.